### DISCUSSION TITLE
How to use lego in Microsoft Windows

### DISCUSSION THREAD

**Discussion by plethora9** [go-acme/lego#1934] [Q&A] [answered]
I downloaded the windows amd64 binary, but how do I use it?  I cannot find documentation on this anywhere.  It's not the same as Linux.  In Windows, we can't start the command with the DNS API token or key.  I don't see a way to run this in command prompt or powershell.

**Comment by ldez** [comment]
Hello,

You have to download and unzip the archive: 
https://github.com/go-acme/lego/releases/download/v4.12.1/lego_v4.12.1_windows_amd64.zip

You will have the binary (`lego.exe`) and license.

After you just have to follow the documentation https://go-acme.github.io/lego/usage/cli/ to run the binary (`lego.exe`)

**Reply by plethora9** [reply]
Hello, thanks for the reply.  I did see that link on the documentation but it does not cover Windows usage.  It only covers how to do it in Linux where it is possible to specify a parameter before the lego command.  Windows won't let me do that.

**Reply by ldez** [reply]
There are no specific things for Windows except the binary name (`.exe`) and path separator (`\`).
Windows lets you use CLI flags: lego is a command line, you have to run it inside a terminal.

**Reply by plethora9** [reply]
Sorry it is still not working or I am still not understanding.  I am trying to run this from an administrator command prompt on Windows Server 2019.  It is not letting me set the flag for the DNS API Token.
![Xi3f67B](https://github.com/go-acme/lego/assets/5674651/15ffceb6-d605-4c26-8661-93696c27279e)
![qGW3UP3-i](https://github.com/go-acme/lego/assets/5674651/633d77bf-0e01-42ff-a014-9d24d3992025)

**Reply by ldez** [reply, marked as answer]
You have to set the environment variables (they are not CLI flags) with your credentials.
Then you cannot do `-CLOUDFLARE_DNS_API_TOKEN=xxx`.

You have to remove `\` at the end of the lines.

```bash
CLOUDFLARE_DNS_API_TOKEN=1234567890abcdefghijklmnopqrstuvwxyz lego --email you@example.com --dns cloudflare --domains my.example.org run
```

another way:

```shell
setx CLOUDFLARE_DNS_API_TOKEN "1234567890abcdefghijklmnopqrstuvwxyz"
lego --email you@example.com --dns cloudflare --domains my.example.org run
```

NOTE: I recommend using something better than the default Windows terminal (like Git BASH or WSL)

**Reply by plethora9** [reply]
Got it, thank you!  That second method worked.  Here is what I ended up with for now.  I will check out BASH and WSL as soon as I can.  Thanks again!

`SET CLOUDFLARE_DNS_API_TOKEN=1234567890abcdefjhijklmnopqrstuvwxyz && lego --email user@domain.com --dns cloudflare --key-type rsa2048 -d device.domain.com --accept-tos --pem run`

I removed the path parameter and just let it use the .lego sub-folder.

Also thanks for fixing my second screenshot :)
